exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -1949,3 +2031,34 @@ subSort:
 		break
 	}
 }
+
+func assetPairFromSymbol(symbol string) (asset.Item, currency.Pair, error) {
+	assetType := asset.Spot
+
+	if symbol == "" {
+		return assetType, currency.EMPTYPAIR, nil
+	}
+
+	pairInfo := strings.Split(symbol, ":")
+	switch len(pairInfo) {
+	case 1:
+		newPair := pairInfo[0]
+		if newPair[0] == 'f' {
+			assetType = asset.MarginFunding
+		}
+		symbol = newPair[1:]
+	case 2:
+		assetType = asset.Margin
+		symbol = symbol[1:]
```

Comment:
Note that a spot/margin asset can both contain a ":" delimiter. A pattern I've noticed for bitfinex spot/margin pairs is that anything where the base + quote len > 6, requires it. With derivatives, the pair always has one but GCT does not support it on websocket. As it stands, any pair with a len > 6 will throw a subscription issue so I've made a patch to address this.

[1.patch](https://github.com/thrasher-corp/gocryptotrader/files/12450131/1.patch)

Next up, with the delimiter, this would return an enabled spot pair like DOGE:USD or LUNA2:USD as a margin asset. As a separate issue with this, margin assets with a len of 6 like IOTUSD would not contain a delimiter in its returned symbol which this would currently report it as a spot asset.

Reply:
Thanks.
I've applied the patch and it makes sense.
Looking into assetPairFromSymbol now.

Reply:
Okay. Think that's solved it.
Tests added.

Reply:
Nope. More to do. I'll reply when I'm done.

Reply:
Good to go now. Disabled Margin assets for WS until we solve the collisions, multiplex the responses (🤮 ) or something else.